src/coreclr/scripts/superpmi.py

```diff
@@ -1623,7 +1623,13 @@ def replay(self):
                         logging.warning("Warning: SuperPMI returned a zero exit code, but generated a non-zero-sized mcl file")
                     print_fail_mcl_file_method_numbers(fail_mcl_file)
                     repro_base_command_line = "{} {} {}".format(self.superpmi_path, " ".join(repro_flags), self.jit_path)
-                    save_repro_mc_files(temp_location, self.coreclr_args, repro_base_command_line)
+
+                    # Use the MCH file base name as the main part of the directory for generated repro artifacts.
+                    # Remove the trailing ".mch", if any.
+                    artifacts_base_name = os.path.basename(mch_file)
+                    artifacts_base_name = artifacts_base_name[:-4] if artifacts_base_name.lower().endswith(".mch") else artifacts_base_name
```

Comment:
May be repurpose `-tag` for these files as well? It might turn out useful.

Reply:
It seems unlikely someone would specify `-tag` for a non-asmdiffs replay just to put the repro mcs in a differently named directory. I never even use the repro MC files, myself. I'm not sure I want complicate (further) the options to support that.

Reply:
sounds reasonable.